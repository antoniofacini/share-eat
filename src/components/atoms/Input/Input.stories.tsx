import { Story, Meta } from '@storybook/react/types-6-0'
import Input from '.'

export default {
  title: 'Atoms/Input',
  component: Input,
} as Meta

export const Default: Story = (args) => (
  <Input
    error
    errorMessage={'Mensagem de erro'}
    fieldName="Nome"
    label="Nome do prato"
    placeholder="Prato"
    register={() => console.log('handle')}
  />
)
