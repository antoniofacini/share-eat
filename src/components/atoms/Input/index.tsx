import { FieldValues, UseFormRegister } from 'react-hook-form'

interface IInput {
  error: boolean
  errorMessage: string | undefined
  fieldName: string
  label: string
  placeholder: string
  // prefix?: string
  tip?: string
  type?: string
  register?: UseFormRegister<FieldValues> | any
  dataTestId?: string
}

const Input = ({
  placeholder = 'Descrição',
  // prefix = '',
  type = 'text',
  fieldName = 'field',
  label = 'Descrição',
  error = false,
  errorMessage = 'Verifique as informações inseridas',
  tip = '',
  register,
  dataTestId,
}: IInput) => (
  <div className="grid gap-1">
    <label
      htmlFor={fieldName}
      className="w-full text-start text-sm font-bold font-sans text-white-blur p-1"
    >
      {label}
    </label>
    <input
      id={fieldName}
      type={type}
      placeholder={placeholder}
      className="w-full rounded p-3"
      data-test-id={dataTestId}
      {...register}
    />
    {tip && !error && <p className="text-white-blur font-light text-sm tracking-tighter">{tip}</p>}
    {error && <p className="w-full text-start text-sm text-white p-1">{errorMessage}</p>}
  </div>
)

export default Input
