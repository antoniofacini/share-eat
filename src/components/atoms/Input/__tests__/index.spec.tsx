import { render } from '@testing-library/react'
import Input from '..'

describe('<Input />', () => {
  it('should render Input component', () => {
    const wrapper = render(
      <Input
        error
        errorMessage="Mensagem de erro"
        fieldName="Nome"
        label="Nome do prato"
        placeholder="Prato"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
