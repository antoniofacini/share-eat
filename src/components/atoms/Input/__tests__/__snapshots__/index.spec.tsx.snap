// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render Input component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid gap-1"
      >
        <label
          class="w-full text-start text-sm font-bold font-sans text-white-blur p-1"
          for="Nome"
        >
          Nome do prato
        </label>
        <input
          class="w-full rounded p-3"
          id="Nome"
          placeholder="Prato"
          type="text"
        />
        <p
          class="w-full text-start text-sm text-white p-1"
        >
          Mensagem de erro
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid gap-1"
    >
      <label
        class="w-full text-start text-sm font-bold font-sans text-white-blur p-1"
        for="Nome"
      >
        Nome do prato
      </label>
      <input
        class="w-full rounded p-3"
        id="Nome"
        placeholder="Prato"
        type="text"
      />
      <p
        class="w-full text-start text-sm text-white p-1"
      >
        Mensagem de erro
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
